uwblueprint/website-bp
# Fix ESLint and add React rules

### Implementation description

This updates our eslint config to 1) work and 2) use recommended ESLint rules to avoid common and critical pitfalls surrounding React hooks.

This should have no visible impacts on the site.  

### Steps to test

1. Run `yarn dev` and ensure nothing is on fire
2. Try calling a react hook conditionally.  You should see an error appear in your IDE. (Hooks cannot be called conditionally - this is a core invariant of React.)

### Focus areas
See PR comments <3
